### PR TITLE
Feature: Add action buttons outside card (kids-1763)

### PR DESCRIPTION
**Pull request by Daniela510**
## Summary by CodeRabbit

- **New Features**
	- Added new analytics event identifiers for tracking user interactions.
	- Introduced a `ScrollController` for improved scrolling behavior in the Grateful Screen.
	- Enhanced the Recommendations List with new buttons for user interactions, including a skip option.
	- Added functionality to conditionally skip generous deeds based on user interactions.

- **Bug Fixes**
	- Updated event handling to streamline analytics data capture and improve error handling in user interactions.

- **Chores**
	- Improved resource management by ensuring proper disposal of the scroll controller.

### CODE REVIEW

File: lib/features/family/features/reflect/bloc/grateful_cubit.dart [hunk 1]
```diff
@@ -172,6 +172,10 @@ class GratefulCubit extends CommonCubit<GratefulUIModel, GratefulCustom> {
   }
 
   void onRecommendationChosen(int index) {
+    if (index == -1) {
```
Comment (TammiLion) [inline lib/features/family/features/reflect/bloc/grateful_cubit.dart:175]: Why can the index be -1? In what situation does this occur?

File: lib/features/family/features/reflect/presentation/widgets/recommendations_list_widget.dart [hunk 2]
```diff
@@ -128,6 +137,39 @@ class _RecommendationsListWidgetState extends State<RecommendationsListWidget> {
...
+                            ? 'act_of_service'
+                            : 'donation':
+                        widget.uiModel.organisations[_currentIndex].name,
+                  }),
+            ),
+          ),
+        const SizedBox(height: 16),
+        TextButton(
+          onPressed: () {
+            widget.onRecommendationChosen?.call(-1);
```
Comment (TammiLion) [inline lib/features/family/features/reflect/presentation/widgets/recommendations_list_widget.dart:163]: Ah I see now where the -1 comes from. I would not do it this way. I recommend making a new fuction param in RecommendationsListWidget onSkip, and then cubit.onSkip. Then in cubit.onSkip you can still make use of onDeed.

**Review by TammiLion — CHANGES_REQUESTED** [review]
See comments about the -1 index

File: lib/features/family/features/reflect/presentation/widgets/recommendations_list_widget.dart (hunk 2 above)
Comment (Daniela510) [reply lib/features/family/features/reflect/presentation/widgets/recommendations_list_widget.dart]: good point!

**Review by TammiLion — APPROVED** [review]

**Pull request merged**